### PR TITLE
Add feedback on item darg & drop

### LINKED ISSUE #2685: [regionOfInterest] missing signal 'editingFinished' ?
It would be interesting to have an 'editingFinished' signal for RegionOfInterest.

For example in the case of computing statistics on specific plot item and region of interest, computation can be time consuming. So if we compute it for each new anchor position it will be a nightmare.
We should only recompute those when the edition is finished.
Or maybe the emission of `regionChanged` signal can be changed.

### PR DESCRIPTION
**Pull request by payno**
/close #2685

### CODE REVIEW

**Comment by vallsv** [comment]
Is it part of the release?

**Comment by payno** [comment]
nope

File: silx/gui/plot/items/roi.py [hunk 1]
```diff
@@ -113,6 +120,7 @@ def __init__(self, parent=None):
         self._editable = False
         self._visible = True
         self.sigItemChanged.connect(self.__itemChanged)
+        self._editing = False
```
Comment (t20100) [inline silx/gui/plot/items/roi.py]: `_editing` can be removed, it is not used.

File: silx/gui/plot/items/marker.py [hunk 2]
```diff
@@ -167,6 +172,12 @@ def _defaultConstraint(*args):
         """Default constraint not doing anything"""
         return args
 
+    def startDrag(self):
```
Comment (t20100) [inline silx/gui/plot/items/marker.py]: Maybe rename `start|endDrag` to `_start|_endDrag`, we don't want someone to call them directly.

File: silx/gui/plot/items/marker.py [hunk 3]
```diff
@@ -35,14 +35,19 @@
...
-
+from silx.gui import qt
 
 _logger = logging.getLogger(__name__)
 
 
 class MarkerBase(Item, DraggableMixIn, ColorMixIn, YAxisMixIn):
     """Base class for markers"""
 
+    sigItemPressed = qt.Signal()
```
Comment (t20100) [inline silx/gui/plot/items/marker.py]: I'd prefer `sigDragStarted|Finished`.

File: silx/gui/plot/items/roi.py [hunk 4]
```diff
@@ -100,8 +100,15 @@ class RegionOfInterest(_RegionOfInterestBase):
     sigRegionChanged = qt.Signal()
     """Signal emitted everytime the shape or position of the ROI changes"""
 
+    sigRegionEditionStarted = qt.Signal()
```
Comment (t20100) [inline silx/gui/plot/items/roi.py]: I'd prefer `sigDragStarted|Finished`

File: silx/gui/plot/items/roi.py [hunk 5]
```diff
@@ -505,6 +519,15 @@ def __str__(self):
         params = '; '.join('(%f; %f)' % (pt[0], pt[1]) for pt in points)
         return "%s(%s)" % (self.__class__.__name__, params)
 
+    def _startEditing(self, *args, **kwargs):
```
Comment (t20100) [inline silx/gui/plot/items/roi.py]: Can be renamed `_start|_endDrag`

File: silx/gui/plot/items/roi.py (hunk 4 above)
Comment (t20100) [reply silx/gui/plot/items/roi.py]: ....Though it's not exactly a drag of the whole ROI.... to see.

**Comment by t20100** [comment]
Sounds like missing signals, indeed.

I put a few comments in the review, main one is: I'd rename `MarkerBase.sigItemPressed|Released` and `RegionOfInterest.sigRegionEditionStarted|Finished` to `MarkerBase.sigDragStarted|Finished` as it is more meaningful and makes the API consistent between markers and ROIs (and similar to QtQuick API https://doc.qt.io/qt-5/qml-qtquick-drag.html#attached-signals).

File: silx/gui/plot/PlotInteraction.py [hunk 6]
```diff
@@ -1263,6 +1263,8 @@ def beginDrag(self, x, y):
 
         if isinstance(item, items.MarkerBase):
             self._signalMarkerMovingEvent('markerMoving', item, x, y)
+            if item is not None:
```
Comment (t20100) [inline silx/gui/plot/PlotInteraction.py]: BTW, `item` cannot be `None`, it is an instance of `MarkerBase`

File: silx/gui/plot/items/roi.py [hunk 7]
```diff
@@ -392,8 +399,14 @@ def _createPlotItems(self):
                 item.setColor(color)
                 item.setVisible(self.isVisible())
                 plot._add(item)
+                # connect item changed
                 item.sigItemChanged.connect(functools.partial(
                     self._controlPointAnchorChanged, index))
+                # connect items pressed and released signals
+                item.sigDragStarted.connect(functools.partial(
```
Comment (t20100) [inline silx/gui/plot/items/roi.py]: One final comment, the `functools.partial` is actually not used as the parameters of `_editingStarted|Finished` are not used, probably best to remove unless you want to give the index in `sigEditingStarted|Finished` which can be a good idea if it is meaningful

File: silx/gui/plot/items/roi.py [hunk 8]
```diff
@@ -505,6 +518,13 @@ def __str__(self):
         params = '; '.join('(%f; %f)' % (pt[0], pt[1]) for pt in points)
         return "%s(%s)" % (self.__class__.__name__, params)
 
+    def _editingStarted(self, *args, **kwargs):
```
Comment (t20100) [inline silx/gui/plot/items/roi.py]: `*args, **kwargs` is not used.

**Comment by t20100** [comment]
LGTM

**Pull request merged**